tests/h/schemas/annotation_test.py

```diff
@@ -608,6 +613,70 @@ def test_it_passes_existing_target_uri_to_document_metas_from_data(
             claimant=mock.sentinel.target_uri)
 
 
+class TestSearchParamsSchema(object):
+
+    def test_it_returns_known_params(self, schema):
+        input_params = {}
+
+        params = schema.validate(input_params)
+
+        assert params == input_params
+
+    def test_it_ignores_unknown_params(self, schema):
+        expected_params = {
+            '_separate_replies': '1',
+            'created': "20180814",  # yyyyMMdd
+            'group': "group1",
+            'quote': "quote me",
+            'references': "3456TA12",
+            'tag': "tagme",
+            'tags': "[tagme2]",
+            'text': "text me",
+            'target': "link me",
+            'thread_ids': "threads",
+            'updated': "20180814",  # yyyyMMdd
+            'uri': "foobar.com",
+            'url': "https://foobar.com",
+            'any': "foo",
+            'user': "pooky",
+            'sort': "created",
+            'limit': "10",
+            'order': "asc",
+            'offset': "0",
+        }
+        unknown_params = {
+            "paramdoesnotexist": "value",
+            "paramnoexiste": "value"}
+        input_params = deepcopy(expected_params)
+        input_params.update(unknown_params)
+
+        params = schema.validate(input_params)
+
+        assert params == expected_params
+
+    def test_it_handles_duplicate_keys(self, schema):
+        expected_params = MultiDict([
+            ("url", "http://foobar"),
+            ("url", "http://foobat")])
+        input_params = deepcopy(expected_params)
+        input_params.add("unknownparam", "foo")
+        input_params.add("unknownparam", "bar")
+
+        params = schema.validate(input_params)
+
+        assert params == expected_params
+
+    def test_raises_if_invalid_sorting_order(self, schema):
```

Comment:
I'm thinking if we do this we may need to handle this "nicely" in the client and activity pages?